File: Old JAVA/model/GroupAlim.java
package model;

import java.util.ArrayList;

public enum GroupAlim {
	ALL("Tous", 1,0, targetAdjust.NO,999),
ABATS("Abats", 1,0, targetAdjust.PROT,0),
AIDE("Aides culinaires", 10,0, targetAdjust.NO,1),
ALGUES("Algues", 5,0, targetAdjust.NO,2),
AROMATEf("Aromates frais",5,0, targetAdjust.NO,3),
AROMATEs("Aromates séchés",5,0, targetAdjust.NO,4),
AutreCereal("Autres céréales",3,0, targetAdjust.ENERGIE,5),
AutrePOISSON("Autres produits à base de poisson",1,0, targetAdjust.PROT,6),
ALIMenf("Aliments infantiles",10,0, targetAdjust.ENERGIE,7),
MGAnim("Autres matières grasses animales", 4,0, targetAdjust.LIP,8),
MGLait("Beurres et matières grasses laitières",4,0, targetAdjust.LIP,9),
MGAutre("Margarines et matières grasses composées",4,0, targetAdjust.LIP,10),
BISCOTTE("Biscottes et pains non levés", 3,0, targetAdjust.ENERGIE,11),
BISCUITSAL("Biscuits salés apéritifs", 3,0, targetAdjust.ENERGIE,12),
BISCUITScu("Biscuits secs sucrés",3,0, targetAdjust.ENERGIE,13),
BOUILLON("Bouillons préts à consommer", 3,0, targetAdjust.ENERGIE,14),
CEREALPD("Céréales petit déjeuner et barres céréalières",3,0, targetAdjust.ENERGIE,15),
CHARCUT("Charcuteries et salaisons", 1,0, targetAdjust.PROT,16),
COMPLEMENT("Compléments alimentaires",5,0, targetAdjust.CALCIUMPHOS,17),
CREMES("Crèmes et spécialités à base de créme", 1,0, targetAdjust.PROT,18),
CRUSTACE("Crustacés et mollusques",1,0, targetAdjust.PROT,19),
DESSERTLAIT("Desserts lactés", 1,0, targetAdjust.PROT,20),
DESSERT("Desserts Autres",3,0, targetAdjust.ENERGIE,21),
EAUX("Eaux", 6,0, targetAdjust.NO,22),
EPICES("Epices",10,0, targetAdjust.NO,23),
FRUITSIRP("Fruits au sirop/au jus, compotes", 2,0, targetAdjust.NO,24),
FRUITFRAIS("Fruits", 2,0, targetAdjust.NO,25),
FRUITCOQUE("Fruits à coques et graines oléagineuses",4,0, targetAdjust.NO,26),
JUS("Jus", 10,0, targetAdjust.NO,27),
AutreFRUITS("Autres produits transformés à base de fruits",2,0, targetAdjust.NO,28),
FarinesAmidon("Farines et amidons",3,0, targetAdjust.ENERGIE,29),
PAINS("Pains",3,0, targetAdjust.ENERGIE,30),
VIENNOISERIE("Viennoiseries et brioches",3,0, targetAdjust.ENERGIE,31),
GATEAUX("Gateaux et patisseries",3,0, targetAdjust.ENERGIE,32),
CAKES("Feuilletés et cakes salés",3,0, targetAdjust.ENERGIE,33),
PATETARTE("Pâtes à tarte, à pizza, etc.",3,0, targetAdjust.ENERGIE,34),
FROMAGES("Fromages", 1,0, targetAdjust.PROT,35),
FRUITSEC("Fruits séchés ou lyophilisés", 2,0, targetAdjust.NO,36),
EPICE("Herbes, épices et assaisonnements",5,0, targetAdjust.NO,37),
HUILES("Huiles et graisses végétales", 4,0, targetAdjust.LIP,38),
LAITS("Laits", 1,0, targetAdjust.PROT,39),
LEGUMES("Légumes",2,0, targetAdjust.FIBER,40),
DIVERS("Ingrédients divers",10,0, targetAdjust.NO,41),
LEGUMESSEC("Légumes secs", 2,0, targetAdjust.FIBER,42),
OEUF("Oeufs et dérivés",1,0, targetAdjust.PROT,43),
HUILEPOISSON("Huiles de poissons",4,0, targetAdjust.EPA,44),
POISSON("Poissons et batraciens",1,0, targetAdjust.PROT,45),
POMMETERRE("Pommes de terre et apparentés", 3,0, targetAdjust.ENERGIE,46),
PATES("Pâtes et semoules", 3,0, targetAdjust.ENERGIE,47),
SAUCES("Sauces et condiments",10,0, targetAdjust.NO,48),
SELS("Sels",5,0, targetAdjust.NA,49),
POISSONPROD("Produits à base de poissons", 1,0, targetAdjust.PROT,50),
RIZ("Riz et autres graines",3,0, targetAdjust.ENERGIE,51),
SALADE("Salades composées et crudités",2,0, targetAdjust.FIBER,52),
PLAT("Plats composés",10,0, targetAdjust.NO,53),
SANDWICH("Sandwichs",10,0, targetAdjust.NO,54),
SOUPE("Soupes, Bouillons",10,0, targetAdjust.NO, 55),
SUCRE("Sucres, miels, sirops, confiseries", 5,0, targetAdjust.ENERGIE,56),
VIANDES("Viandes", 1,0, targetAdjust.PROT,57),
VOLLAILLE("Volailles", 1,0, targetAdjust.PROT,58),
YAOURT("Yaourts et spécialités laitiéres type yaourts",1,0, targetAdjust.PROT,59),
AUTRES("Autres",10,0, targetAdjust.NO,60),
FLAIT("Produits laitiers et d'oeufs",1,1, targetAdjust.PROT,61),
FARO("Épices et fines herbes",5,1, targetAdjust.NO,62),
FBEBE("Aliments pour bébés",10,1, targetAdjust.NO,63),
FHUILE("Matières grasses et huiles",4,1, targetAdjust.LIP,64),
FVOL("Produits de volaille",1,1, targetAdjust.PROT,65),
FPOT("Potages et sauces",2,1, targetAdjust.LIP,66),
FSAUCI("Saucisses et viandes froides",1,1, targetAdjust.PROT,67),
FCERE("Céréales à déjeuner",3,1, targetAdjust.ENERGIE,68),
FFRUIT("Fruits et jus de fruits",2,1, targetAdjust.FIBER,69),
FPORC("Produits de porc",1,1, targetAdjust.PROT,70),
FLEG("Légumes et produits végétaux",2,1, targetAdjust.FIBER,71),
FNOIX("Noix et graines",3,1, targetAdjust.NO,72),
FBOEUF("Produits de boeuf",1,1, targetAdjust.PROT,73),
FBOISS("Boissons",6,1, targetAdjust.NO, 74),
FPOISS("Produits de poissons, mollusques et crustacés",1,1, targetAdjust.PROT,75),
FLEGUMUNEUSE("Légumineuses et produits de légumineuses",2,1, targetAdjust.FIBER,76),
FAGNEAU("Ageau, veau et gibier",1,1, targetAdjust.PROT,77),
FBOULAN("Produits de boulangerie",3,1, targetAdjust.ENERGIE,78),
FSUCRE("Sucreries",3,1, targetAdjust.ENERGIE,79),
FPAIN("Céréales, grains et pâtes",3,1, targetAdjust.ENERGIE,80),
FPRET("Aliments préts-à-manger",10,1, targetAdjust.NO, 81),
FCOMP("Mets composés",10,1, targetAdjust.ENERGIE,82),
FGRI("Grignotises",10,1, targetAdjust.ENERGIE,83)


	;


private String name = "";


private int categorie=0;
private int type=0;
private targetAdjust target=targetAdjust.ENERGIE;
private int id=0;

//Constructeur
 GroupAlim(String name, int categrorieo, int type, targetAdjust t, int id){
  this.name = name;
this.id=id;
  this.categorie=categrorieo;
 this.type=type;
 this.target=t;
}
 
 public String nameToString() {
	return name;
}
public int getCategorie() {
	return categorie;
}
public static boolean isPresent(GroupAlim sz){
	boolean r=false;
	for (GroupAlim ga: GroupAlim.values()){
		if (ga.equals(sz)){
			r=true;
		}
		
	}
	return r;
}
 public int getId() {
	return id;
}
 public static GroupAlim setById(int id) {
	 GroupAlim r=GroupAlim.AUTRES;
		for (GroupAlim ga: GroupAlim.values()){
			if (ga.getId()==id){
				return ga;
			}
		}
		return r;
 }
public int getType() {
	return type;
}
public static GroupAlim StringToGroup(String sz){
	GroupAlim r=GroupAlim.AUTRES;
	for (GroupAlim ga: GroupAlim.values()){
		if (ga.nameToString().toLowerCase().equals(sz.toLowerCase())){
			return ga;
		}
		
	}
	return r;
}
public static ArrayList<String> ListOfGroup(int sz){

	 ArrayList<String> list =new  ArrayList<String>();
	for (GroupAlim ga: GroupAlim.values()){
		if (ga.getType()==sz){
			list.add(ga.nameToString());
		}
		
	}
	return list;
}
public static ArrayList<GroupAlim> valuesExcept() {
	
	ArrayList<GroupAlim>es=new ArrayList<GroupAlim>();
	for (GroupAlim e:GroupAlim.values()) {
		if (e!=GroupAlim.ALL) {
			es.add(e);
		}
			
	}
	return es;
}
public targetAdjust getTarget() {
	return target;
}
@Override
public String toString() {
	return name;
}
	
}

File: composeApp/src/commonMain/kotlin/fr/vetbrain/vetnutri_mp/Enumerise/GroupAlim.kt
package fr.vetbrain.vetnutri_mp.Enumerise

import fr.vetbrain.vetnutri_mp.Data.Labelable

enum class GroupAlim(
        val categorie: Int,
        val type: Int,
        val target: TargetAdjust,
        val id: Int,
        override val label: String?
) : Labelable {
    ALL(1, 0, TargetAdjust.NO, 999, "all"),
    ABATS(1, 0, TargetAdjust.PROT, 0, "offal"),
    AIDE(10, 0, TargetAdjust.NO, 1, "culinaryAids"),
    ALGUES(5, 0, TargetAdjust.NO, 2, "seaweed"),
    AROMATEf(5, 0, TargetAdjust.NO, 3, "freshHerbs"),
    AROMATEs(5, 0, TargetAdjust.NO, 4, "driedHerbs"),
    AutreCereal(3, 0, TargetAdjust.ENERGIE, 5, "otherCereals"),
    AutrePOISSON(1, 0, TargetAdjust.PROT, 6, "otherFishProducts"),
    ALIMenf(10, 0, TargetAdjust.ENERGIE, 7, "babyFood"),
    MGAnim(4, 0, TargetAdjust.LIP, 8, "animalFats"),
    MGLait(4, 0, TargetAdjust.LIP, 9, "dairyFats"),
    MGAutre(4, 0, TargetAdjust.LIP, 10, "otherFats"),
    BISCOTTE(3, 0, TargetAdjust.ENERGIE, 11, "rusks"),
    BISCUITSAL(3, 0, TargetAdjust.ENERGIE, 12, "savoryBiscuits"),
    BISCUITScu(3, 0, TargetAdjust.ENERGIE, 13, "sweetBiscuits"),
    BOUILLON(3, 0, TargetAdjust.ENERGIE, 14, "readyBroth"),
    CEREALPD(3, 0, TargetAdjust.ENERGIE, 15, "breakfastCereals"),
    CHARCUT(1, 0, TargetAdjust.PROT, 16, "delicatessen"),
    COMPLEMENT(5, 0, TargetAdjust.CALCIUMPHOS, 17, "supplements"),
    CREMES(1, 0, TargetAdjust.PROT, 18, "creams"),
    CRUSTACE(1, 0, TargetAdjust.PROT, 19, "shellfish"),
    DESSERTLAIT(1, 0, TargetAdjust.PROT, 20, "dairyDesserts"),
    DESSERT(3, 0, TargetAdjust.ENERGIE, 21, "otherDesserts"),
    EAUX(6, 0, TargetAdjust.NO, 22, "water"),
    EPICES(10, 0, TargetAdjust.NO, 23, "spices"),
    FRUITSIRP(2, 0, TargetAdjust.NO, 24, "cannedFruits"),
    FRUITFRAIS(2, 0, TargetAdjust.NO, 25, "freshFruits"),
    FRUITCOQUE(4, 0, TargetAdjust.NO, 26, "nuts"),
    JUS(10, 0, TargetAdjust.NO, 27, "juices"),
    AutreFRUITS(2, 0, TargetAdjust.NO, 28, "otherFruits"),
    FarinesAmidon(3, 0, TargetAdjust.ENERGIE, 29, "flourStarch"),
    PAINS(3, 0, TargetAdjust.ENERGIE, 30, "bread"),
    VIENNOISERIE(3, 0, TargetAdjust.ENERGIE, 31, "pastries"),
    GATEAUX(3, 0, TargetAdjust.ENERGIE, 32, "cakes"),
    CAKES(3, 0, TargetAdjust.ENERGIE, 33, "savoryCakes"),
    PATETARTE(3, 0, TargetAdjust.ENERGIE, 34, "dough"),
    FROMAGES(1, 0, TargetAdjust.PROT, 35, "cheese"),
    FRUITSEC(2, 0, TargetAdjust.NO, 36, "driedFruits"),
    EPICE(5, 0, TargetAdjust.NO, 37, "seasonings"),
    HUILES(4, 0, TargetAdjust.LIP, 38, "oils"),
    LAITS(1, 0, TargetAdjust.PROT, 39, "milk"),
    LEGUMES(2, 0, TargetAdjust.FIBER, 40, "vegetables"),
    DIVERS(10, 0, TargetAdjust.NO, 41, "miscellaneous"),
    LEGUMESSEC(2, 0, TargetAdjust.FIBER, 42, "driedVegetables"),
    OEUF(1, 0, TargetAdjust.PROT, 43, "eggs"),
    HUILEPOISSON(4, 0, TargetAdjust.EPA, 44, "fishOils"),
    POISSON(1, 0, TargetAdjust.PROT, 45, "fish"),
    POMMETERRE(3, 0, TargetAdjust.ENERGIE, 46, "potatoes"),
    PATES(3, 0, TargetAdjust.ENERGIE, 47, "pasta"),
    SAUCES(10, 0, TargetAdjust.NO, 48, "sauces"),
    SELS(5, 0, TargetAdjust.NA, 49, "salt"),
    POISSONPROD(1, 0, TargetAdjust.PROT, 50, "fishProducts"),
    RIZ(3, 0, TargetAdjust.ENERGIE, 51, "rice"),
    SALADE(2, 0, TargetAdjust.FIBER, 52, "salads"),
    PLAT(10, 0, TargetAdjust.NO, 53, "dishes"),
    SANDWICH(10, 0, TargetAdjust.NO, 54, "sandwiches"),
    SOUPE(10, 0, TargetAdjust.NO, 55, "soups"),
    SUCRE(5, 0, TargetAdjust.ENERGIE, 56, "sugar"),
    VIANDES(1, 0, TargetAdjust.PROT, 57, "meat"),
    VOLLAILLE(1, 0, TargetAdjust.PROT, 58, "poultry"),
    YAOURT(1, 0, TargetAdjust.PROT, 59, "yogurt"),
    AUTRES(10, 0, TargetAdjust.NO, 60, "others"),
    FLAIT(1, 1, TargetAdjust.PROT, 61, "dairyProducts"),
    FARO(5, 1, TargetAdjust.NO, 62, "herbsSpices"),
    FBEBE(10, 1, TargetAdjust.NO, 63, "babyFood"),
    FHUILE(4, 1, TargetAdjust.LIP, 64, "fatsOils"),
    FVOL(1, 1, TargetAdjust.PROT, 65, "poultryProducts"),
    FPOT(2, 1, TargetAdjust.LIP, 66, "soupsAndSauces"),
    FSAUCI(1, 1, TargetAdjust.PROT, 67, "sausages"),
    FCERE(3, 1, TargetAdjust.ENERGIE, 68, "breakfastCereals"),
    FFRUIT(2, 1, TargetAdjust.FIBER, 69, "fruitsAndJuices"),
    FPORC(1, 1, TargetAdjust.PROT, 70, "porkProducts"),
    FLEG(2, 1, TargetAdjust.FIBER, 71, "vegetableProducts"),
    FNOIX(3, 1, TargetAdjust.NO, 72, "nutsAndSeeds"),
    FBOEUF(1, 1, TargetAdjust.PROT, 73, "beefProducts"),
    FBOISS(6, 1, TargetAdjust.NO, 74, "beverages"),
    FPOISS(1, 1, TargetAdjust.PROT, 75, "seafoodProducts"),
    FLEGUMUNEUSE(2, 1, TargetAdjust.FIBER, 76, "legumes"),
    FAGNEAU(1, 1, TargetAdjust.PROT, 77, "lambAndGame"),
    FBOULAN(3, 1, TargetAdjust.ENERGIE, 78, "bakeryProducts"),
    FSUCRE(3, 1, TargetAdjust.ENERGIE, 79, "sweets"),
    FPAIN(3, 1, TargetAdjust.ENERGIE, 80, "cerealsAndPasta"),
    FPRET(10, 1, TargetAdjust.NO, 81, "readyToEat"),
    FCOMP(10, 1, TargetAdjust.ENERGIE, 82, "composedDishes"),
    FGRI(10, 1, TargetAdjust.ENERGIE, 83, "snacks");

    companion object {
        fun byId(id: Int): GroupAlim = entries.find { it.id == id } ?: AUTRES

        fun isPresent(group: GroupAlim): Boolean = entries.contains(group)

        fun byName(name: String): GroupAlim =
                entries.find { it.label?.equals(name, ignoreCase = true) == true } ?: AUTRES

        fun listByType(type: Int): List<String> =
                entries.filter { it.type == type }.mapNotNull { it.label }

        fun valuesExcept(): List<GroupAlim> = entries.filter { it != ALL }
    }

    override fun toString() = label ?: "Unknown"
}
